### PR TITLE
Remove get_ prefix

### PR DESCRIPTION
**Pull request by tcharding**
This one might be a viewed as code churn or unnecessarily modifying the API, feel free to NACK :)

We have a bunch of methods that use the prefix `get_`, they are not exactly getters because they do more than just access a struct fields so Rust convention relating to getters does not apply, however, the `get_` prefix does not add to the descriptiveness of name hence the shorter form can be used with no loss of clarity.

Improve docs and deprecate any methods changed that are pubic.

### CODE REVIEW

**Comment by dpc** [comment]
BTW. The fact that `.txid()` is so costly always bothered me. Seems to me that some longer name like `.calculate_txid()` etc. would making it more descriptive.

**Comment by apoelstra** [comment]
concept ACK this PR. concept NACK increasing the length of `txid`'s name. I agree it's potentially surprising that this is more than a lookup -- but on the other hand, this is true for a lot of other language bindings (e.g. the Python code in Bitcoin Core's test suite) and not *that* surprising to users who are aware that txids are hashes.

**Comment by Kixunil** [comment]
Definitely ACK removing `get_` - it's Rust idiomatic style.

`txid()` bothers me a bit too. I remember when I first time used it I was asking myself "does it compute the hash or does it cache it in the struct?" and looking into the code. That being said, I do think having a short name is nice.

File: src/blockdata/transaction.rs [hunk 1]
```diff
@@ -416,37 +416,66 @@ impl Transaction {
...
     pub fn get_weight(&self) -> usize {
-        self.get_scaled_size(WITNESS_SCALE_FACTOR)
+        self.weight()
     }
 
-    /// Gets the regular byte-wise consensus-serialized size of this transaction.
+    /// Returns the "weight" of this transaction, as defined by BIP141.
+    ///
+    /// For transactions with an empty witness, this is simply the consensus-serialized size times
+    /// 4. For transactions with a witness, this is the non-witness consensus-serialized size
```
Comment (dr-orlovsky) [inline src/blockdata/transaction.rs:429]: nit: This will format like a numbered list (since the line starts with a number and dot).

**Review by dr-orlovsky — DISMISSED** [review]
ACK c4fb637a58d26d809abeb3367cd658a6997dc71f

Since the old names are deprecated, I assume this can be merged in `0.28` rc fix as well?

File: src/blockdata/transaction.rs (hunk 1 above)
Comment (apoelstra) [reply src/blockdata/transaction.rs:429]: Lol. I guess we should change `4` to `four`.

**Comment by apoelstra** [comment]
And yes, I'm fine bringing this into the RC -- but then all the `since`s need to be updated to say 0.28.0 rather than 0.29.0.

**Review by sanket1729 — DISMISSED** [review]
utACK c4fb637a58d26d809abeb3367cd658a6997dc71f

File: src/blockdata/transaction.rs (hunk 1 above)
Comment (tcharding) [reply src/blockdata/transaction.rs:429]: Ha, that's funny. Good catch @dr-orlovsky

**Comment by tcharding** [comment]
Changes in force push:
- Use 'since 0.28.0 instead of 0.29.0 so we can merge this for the taproot release.
- Used 'four' and 'three' instead of digits to get rendering of rustdocs to work correctly.
- Change comment on `scaled_size` to remove the mention of `get_`.

**Review by apoelstra — DISMISSED** [review]
ACK 6b5f1918578bc115dc3dc6d7f5a7627afca5a329

**Comment by dr-orlovsky** [comment]
Needs rebase

**Comment by tcharding** [comment]
No changes in force push, just rebase.

**Review by dr-orlovsky — APPROVED** [review]
ACK 3bde1a205cb1be0ccc5a4488f9955057fa9c65a0

**Review by sanket1729 — APPROVED** [review]
ACK 3bde1a205cb1be0ccc5a4488f9955057fa9c65a0

**Pull request merged**